banzai_floyds/matched_filter.py

```diff
@@ -350,7 +350,7 @@ def optimize_match_filter(initial_guess, data, error, weights_function, x, weigh
                                      args=(data, error, weights_function,
                                            weights_jacobian_function,
                                            weights_hessian_function, x, *args),
-                                     method='Powell', bounds=bounds)
```

Comment:
I found a rather nasty bug in scipy that using the Powell method with bounds on the parameters can give you the wrong answer even with a good initial guess. What's worse is the success flag on the fit says success even though it definitely isn't. The fitter I've chosen is the default in scipy.minimize.